### DISCUSSION TITLE
core分支下有dev，meta，premium三个目录，也就是说clash有三个版本

### DISCUSSION THREAD

**Discussion by gopherchai** [vernesong/OpenClash#3858] [Q&A]
这三个目录下都有clash的可执行文件，请问它们对应的源码的网址是多少，他们之间区别在有哪些？

**Comment by xiaoyangdkj** [comment]
> 这三个目录下都有clash的可执行文件，请问它们对应的源码的网址是多少，他们之间区别在有哪些？

dev版本（clash-foss）是初始的开源项目

p核是基于f核（dev版本）修改的闭源版本，用于clash for windows和clash for android，相较于f核，多出了节点集、代理集等功能

m核是基于f核（dev版本）修改的开源版本，相较于上面两个的区别是 vless支持，域名嗅探等功能

f核和p核都是同一班人马，m核是clash项目的另一个分支
